locations/spiders/minimart_bg.py

```diff
@@ -0,0 +1,29 @@
+from scrapy import Request, Selector, Spider
+
+from locations.google_url import url_to_coords
+from locations.items import Feature
+
+
+class MinimartBGSpider(Spider):
+    name = "minimart_bg"
+    item_attributes = {"brand": "Minimart", "brand_wikidata": "Q119168386"}
+    allowed_domains = ["mini-mart.bg", "goo.gl"]
```

Comment:
I want no part in spiders that go near Google. But you either need to add google to allowed domains, or just do a head request, without following the redirect and get it from `str(response.headers["Location"])`.

Reply:
```python
    item_attributes = {"brand": "Minimart", "brand_wikidata": "Q119168386"}
    allowed_domains = ["mini-mart.bg", "goo.gl"]
    start_urls = ["https://mini-mart.bg/nameri-magazin/"]
    custom_settings = {"REDIRECT_ENABLED": False}
    handle_httpstatus_list = [302]
    no_refs = True

    def parse(self, response: Response, **kwargs: Any) -> Any:
        for location in response.xpath("//div[contains(@class, 'et_pb_with_border')]"):
            item = Feature()
            item["addr_full"] = location.xpath(".//p//text()").get()
            location_url = location.xpath(".//a/@href").get()
            if location_url is not None:
                yield Request(method="HEAD", url=location_url, meta={"item": item}, callback=self.parse_maps_url)

    # Follow shortened URL to find full Google Maps URL containing coordinates
    def parse_maps_url(self, response):
        item = response.meta["item"]
        item["lat"], item["lon"] = url_to_coords(str(response.headers["Location"]))
        yield item
```

Reply:
Thank you, it is working as expected!

> I want no part in spiders that go near Google

When I ran my initial code locally it fetched the coordinates from Google (with some changes since they seemed very accurate). Now some shops are a bit off (a few blocks at most). I assume that Google is doing something in the background to correct the coordinates and that's why you don't want to go near Google?

Reply:
We don't want to go near Google because we don't want to copy their data or appear like we're copying their data.